#!/usr/bin/env python
PACKAGE = "evarobot_odometry"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("alwaysOn",          bool_t,   0, "if false, it doesn't publish data until a node subscribes",   True)
gen.add("wheelSeparation",   double_t, 0, "Difference between wheel in meters",                          0.32, 0.0, 1.0)
gen.add("height",            double_t, 0, "Height of Evarobot base_link in meters",     				 0.25, 0.0, 1.0)
gen.add("gearRatio",         int_t,    0, "Gear Ratio of motors",   									 50, 0, 200)
gen.add("CPR",               int_t,    0, "CPR value of encoders.", 									 64, 0, 100)
gen.add("wheelDiameter",     double_t, 0, "Diameter of driver wheel in meter",    						 0.17, 0.0, 1.0)

exit(gen.generate(PACKAGE, "evarobot_odometry", "Params"))
